lib/utils/search.js

```diff
@@ -32,12 +38,47 @@ const getWeight = (searchTerm, words, resource, resourceIndex) => {
   return weight;
 };
 
+const weightByCategories = (ids, items) => {
+  return JSON.parse(JSON.stringify(items)).map(item => {
+    item.resources = item.resources.map(resource => {
+      resource.weight = getCategoryWeight(ids, items, resource);
+      return resource;
+    });
+    return item;
+  });
+};
+
+const getCategoryWeight = (ids, categorisedResources, resource) => {
+  const categoriesResourceIsIn = categorisedResources
+    .filter(category => {
+      return category.resources.some(r => r.id == resource.id);
+    })
+    .map(x => x.id);
+
+  const selectedCategoriesResourceIsIn = ids.filter(id => {
+    return categoriesResourceIsIn.includes(id);
+  });
+
+  const unselectedCategoriesResourceIsIn = categoriesResourceIsIn.filter(id => {
+    return !ids.includes(id);
+  });
+
+  let percentageMatch = (selectedCategoriesResourceIsIn.length / ids.length) * 100;
+
+  unselectedCategoriesResourceIsIn.forEach(e => {
+    percentageMatch = percentageMatch * 0.8;
+  });
+
+  return percentageMatch | 0;
```

Comment:
What is | ?

Reply:
It should convert percentageMatch to an integer.

Reply:
ooooh